### PR TITLE
Align more DataFrame APIs with pandas

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
I have a script that did some signature comparisons between `pandas.DataFrame` and `cudf.DataFrame` API and it appears some signatures have changed between the pandas 1.x and 2.x release. The API changes in this PR are mostly adding implementations or adding missing keyword argument (although they might not be implemented). The APIs affected are:

* `__init__`
* `__array__`
* `__arrow_c_stream__`
* `to_dict`
* `where`
* `add_prefix`
* `join`
* `apply`
* `to_records`
* `from_records`
* `unstack`
* `pct_change`
* `sort_values`

Marking as breaking as I ensured some added keywords are in the same positions as pandas and therefore might break users who are using purely positional arguments.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -1538,6 +1539,25 @@ def __array_function__(self, func, types, args, kwargs):
...
+            PyCapsule containing a C ArrowSchema representation of the
+            requested schema. Currently not implemented.
+
+        Returns
+        -------
+        PyCapsule
+        """
+        if requested_schema is not None:
+            raise NotImplementedError("requested_schema is not supported")
+        return self.to_arrow().__arrow_c_stream__()
```
Comment (mroeschke) [inline python/cudf/cudf/core/dataframe.py:1559]: If there's more native way to do this that doesn't rely on pyarrow happy to use that

File: python/cudf/cudf/core/dataframe.py [hunk 2]
```diff
@@ -695,6 +695,7 @@ def __init__(
         index=None,
         columns=None,
         dtype=None,
+        copy=None,
```
Comment (galipremsagar) [inline python/cudf/cudf/core/dataframe.py:698]: Is `copy` meant to be non-functional for now? Should we raise NotImplementedError ?

Comment (mroeschke) [reply python/cudf/cudf/core/dataframe.py:701]: Yeah good catch; this is currently not implemented. I'll make it raise a NotImplementedError

**Review by galipremsagar — APPROVED** [review]

**Pull request merged**